scripts/generateTfsdkTags.py

```diff
@@ -7,6 +7,30 @@
 modelFiles = glob.glob('model_*.go')
 jsonRegex = re.compile('`json:\"([0-9A-Za-z]*)[\",]')
 
+def removeUnderscoresFromGroup(group) -> str:
+    return re.sub(r'_', '', group)
+
+def buildAttributeValue(attribute) -> str:
+    underscoreAttribute = ""
+    match = re.findall(r'[0-9A-Z]', attribute)
+    if match != None:
+        underscoreAttribute = re.sub(r'([A-Z])', r'_\g<0>', attribute).lower()
+        fourUpperChars = re.findall(r'[A-Z]{4,4}', attribute)
+        threeUpperChars = re.findall(r'[A-Z]{3,3}', attribute)
+        twoUpperChars = re.findall(r'[A-Z]{2,2}', attribute)
+        if len(fourUpperChars) > 0:
+            underscoreAttribute = re.sub(r'_[a-z]_[a-z]_[a-z]_[a-z]', lambda m: "_" + removeUnderscoresFromGroup(m.group()) + "_", underscoreAttribute).lower()
+        elif len(threeUpperChars) > 0:
+            underscoreAttribute = re.sub(r'_[a-z]_[a-z]_[a-z]', lambda m: "_" + removeUnderscoresFromGroup(m.group(0)[:-1]) + m.group(0)[-1] + "_", underscoreAttribute).lower()
```

Comment:
What are the negative one indexes for here?

Reply:
it's just grouping properly based on the regex => `jwks_url` would be `jwks_url_u_r_l`, `allow_unidentified_client_roc_reds` would be `allow_unidentified_client_roc_r_o_c_reds`